// animal-island-ui/dist/es/index.js
import { jsxs as b, jsx as e, Fragment as X } from "react/jsx-runtime";
import W, { useState as L, useCallback as R, useMemo as aa, useRef as J, useEffect as S } from "react";
import { createPortal as na } from "react-dom";
import ea from "../files/icon-leaf.40329f03.png";
const ta = "animal-btn-QoKe1", ia = "animal-btn-small-6Srsd", oa = "animal-btn-middle-zEXqv", ca = "animal-btn-large-k9IvA", la = "animal-btn-default-L7yE6", sa = "animal-btn-primary-9tMG9", ra = "animal-btn-dashed-wzZ-i", da = "animal-btn-text-Tope4", ma = "animal-btn-link-pURnf", pa = "animal-btn-danger-zQ4kF", ha = "animal-btn-ghost-Ykty4", ua = "animal-btn-block-MC4pp", ba = "animal-btn-loading-PfXLM", fa = "animal-animal-btn-loading-Rf1hg", ga = "animal-btn-icon-P5CS9", M = {
  btn: ta,
  "btn-small": "animal-btn-small-6Srsd",
  btnSmall: ia,
  "btn-middle": "animal-btn-middle-zEXqv",
  btnMiddle: oa,
  "btn-large": "animal-btn-large-k9IvA",
  btnLarge: ca,
  "btn-default": "animal-btn-default-L7yE6",
  btnDefault: la,
  "btn-primary": "animal-btn-primary-9tMG9",
  btnPrimary: sa,
  "btn-dashed": "animal-btn-dashed-wzZ-i",
  btnDashed: ra,
  "btn-text": "animal-btn-text-Tope4",
  btnText: da,
  "btn-link": "animal-btn-link-pURnf",
  btnLink: ma,
  "btn-danger": "animal-btn-danger-zQ4kF",
  btnDanger: pa,
  "btn-ghost": "animal-btn-ghost-Ykty4",
  btnGhost: ha,
  "btn-block": "animal-btn-block-MC4pp",
  btnBlock: ua,
  "btn-loading": "animal-btn-loading-PfXLM",
  btnLoading: ba,
  "animal-btn-loading": "animal-animal-btn-loading-Rf1hg",
  animalBtnLoading: fa,
  "btn-icon": "animal-btn-icon-P5CS9",
  btnIcon: ga
}, q = ({
  type: a = "default",
  size: i = "middle",
  danger: t = !1,
  ghost: o = !1,
  block: c = !1,
  loading: n = !1,
  disabled: l = !1,
  icon: s,
  htmlType: r = "button",
  children: f,
  className: y,
  ...k
}) => {
  const u = [
    M.btn,
    M[`btn-${a}`],
    M[`btn-${i}`],
    t && M["btn-danger"],
    o && M["btn-ghost"],
    c && M["btn-block"],
    n && M["btn-loading"],
    y
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ b(
    "button",
    {
      type: r,
      className: u,
      disabled: l,
      ...k,
      children: [
        s && !n && /* @__PURE__ */ e("span", { className: M["btn-icon"], children: s }),
        f && /* @__PURE__ */ e("span", { children: f })
      ]
    }
  );
};
q.displayName = "Button";
const wa = "animal-wrapper-IRrhj", ya = "animal-wrapper-disabled-MWwHK", Ca = "animal-input-12WUn", ka = "animal-wrapper-small-mWvja", va = "animal-wrapper-middle-zXqMR", Na = "animal-wrapper-large-RrYM4", La = "animal-wrapper-error-W-VRo", Da = "animal-wrapper-warning-miO7A", xa = "animal-prefix-g8f1R", Ia = "animal-suffix-o7ywP", Sa = "animal-clear-HuR2N", B = {
  wrapper: wa,
  "wrapper-disabled": "animal-wrapper-disabled-MWwHK",
  wrapperDisabled: ya,
  input: Ca,
  "wrapper-small": "animal-wrapper-small-mWvja",
  wrapperSmall: ka,
  "wrapper-middle": "animal-wrapper-middle-zXqMR",
  wrapperMiddle: va,
  "wrapper-large": "animal-wrapper-large-RrYM4",
  wrapperLarge: Na,
  "wrapper-error": "animal-wrapper-error-W-VRo",
  wrapperError: La,
  "wrapper-warning": "animal-wrapper-warning-miO7A",
  wrapperWarning: Da,
  prefix: xa,
  suffix: Ia,
  clear: Sa
}, $a = ({
  size: a = "middle",
  prefix: i,
  suffix: t,
  allowClear: o = !1,
  status: c,
  disabled: n = !1,
  className: l,
  value: s,
  defaultValue: r,
  onChange: f,
  onClear: y,
  ...k
}) => {
  const [u, p] = L(r ?? ""), m = s !== void 0, v = m ? s : u, d = R(
    ($) => {
      m || p($.target.value), f?.($);
    },
    [m, f]
  ), C = R(() => {
    m || p(""), y?.();
    const $ = new Event("input", { bubbles: !0 }), h = { value: "" };
    f?.({
      target: h,
      currentTarget: h,
      nativeEvent: $
    });
  }, [m, f, y]), P = [
    B.wrapper,
    B[`wrapper-${a}`],
    c && B[`wrapper-${c}`],
    n && B["wrapper-disabled"],
    l
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ b("span", { className: P, children: [
    i && /* @__PURE__ */ e("span", { className: B.prefix, children: i }),
    /* @__PURE__ */ e(
      "input",
      {
        className: B.input,
        disabled: n,
        value: v,
        onChange: d,
        ...k
      }
    ),
    o && v && !n && /* @__PURE__ */ e(
      "span",
      {
        className: B.clear,
        onClick: C,
        role: "button",
        tabIndex: -1,
        children: "×"
      }
    ),
    t && /* @__PURE__ */ e("span", { className: B.suffix, children: t })
  ] });
};
$a.displayName = "Input";
const Ma = "animal-switch-disabled-U5tjL", Ba = "animal-switch-checked-6HKL4", Aa = "animal-switch-small-jkWtD", Ta = "animal-handle-KgZ7M", Ea = "animal-inner-5x97M", Ra = "animal-switch-loading--3JYL", Pa = "animal-spinner-QK4sP", Ha = "animal-animal-spin-P6YYa", A = {
  switch: "animal-switch-NQPLj",
  "switch-disabled": "animal-switch-disabled-U5tjL",
  switchDisabled: Ma,
  "switch-checked": "animal-switch-checked-6HKL4",
  switchChecked: Ba,
  "switch-small": "animal-switch-small-jkWtD",
  switchSmall: Aa,
  handle: Ta,
  inner: Ea,
  "switch-loading": "animal-switch-loading--3JYL",
  switchLoading: Ra,
  spinner: Pa,
  "animal-spin": "animal-animal-spin-P6YYa",
  animalSpin: Ha
}, Wa = ({
  checked: a,
  defaultChecked: i = !1,
  size: t = "default",
  disabled: o = !1,
  loading: c = !1,
  checkedChildren: n,
  unCheckedChildren: l,
  onChange: s,
  className: r
}) => {
  const [f, y] = L(i), k = a !== void 0, u = k ? a : f, p = R(() => {
    if (o || c) return;
    const v = !u;
    k || y(v), s?.(v);
  }, [o, c, u, k, s]), m = [
    A.switch,
    A[`switch-${t}`],
    u && A["switch-checked"],
    o && A["switch-disabled"],
    c && A["switch-loading"],
    r
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ b(
    "button",
    {
      type: "button",
      role: "switch",
      "aria-checked": u,
      className: m,
      onClick: p,
      disabled: o,
      children: [
        /* @__PURE__ */ e("span", { className: A.handle, children: c && /* @__PURE__ */ e("span", { className: A.spinner }) }),
        /* @__PURE__ */ e("span", { className: A.inner, children: u ? n : l })
      ]
    }
  );
};
Wa.displayName = "Switch";
const z = ({ children: a, className: i, style: t }) => /* @__PURE__ */ e(
  "div",
  {
    className: `animal-cursor${i ? ` ${i}` : ""}`,
    style: t,
    children: a
  }
);
z.displayName = "Cursor";
const Y = (a) => a == null || typeof a == "boolean" ? 0 : typeof a == "string" || typeof a == "number" ? String(a).length : Array.isArray(a) ? a.reduce((i, t) => i + Y(t), 0) : W.isValidElement(a) ? Y(a.props.children) : 0, V = (a, i, t = "tw") => {
  if (i.stopped || a == null || typeof a == "boolean") return null;
  if (typeof a == "string" || typeof a == "number") {
    const o = String(a);
    if (i.remaining >= o.length)
      return i.remaining -= o.length, o;
    const c = o.slice(0, i.remaining);
    return i.remaining = 0, i.stopped = !0, c;
  }
  if (Array.isArray(a))
    return a.map((o, c) => /* @__PURE__ */ e(W.Fragment, { children: V(o, i, `${t}-${c}`) }, `${t}-${c}`));
  if (W.isValidElement(a)) {
    const o = a.props, c = V(o.children, i, t);
    return W.cloneElement(a, void 0, c);
  }
  return null;
}, Q = ({
  children: a,
  speed: i = 90,
  trigger: t,
  autoPlay: o = !0,
  onDone: c
}) => {
  const n = aa(() => Y(a), [a]), [l, s] = L(o ? 0 : n), r = J(null);
  return S(() => {
    if (r.current && window.clearInterval(r.current), !o) {
      s(n);
      return;
    }
    if (s(0), n !== 0)
      return r.current = window.setInterval(() => {
        s((y) => y >= n ? (r.current && window.clearInterval(r.current), y) : y + 1);
      }, i), () => {
        r.current && window.clearInterval(r.current);
      };
  }, [n, i, t, o]), S(() => {
    n > 0 && l >= n && c?.();
  }, [l, n]), /* @__PURE__ */ e(X, { children: V(a, { remaining: l, stopped: !1 }) });
};
Q.displayName = "Typewriter";
const Oa = "animal-mask-hAWeP", ja = "animal-modal-d0zyM", Fa = "animal-modalClipped-7IuAw", qa = "animal-header-udOSi", Ya = "animal-title-s6pv3", Va = "animal-body--aiyx", za = "animal-footer-9Ib-a", T = {
  mask: Oa,
  modal: ja,
  modalClipped: Fa,
  header: qa,
  title: Ya,
  body: Va,
  footer: za
}, Ga = () => /* @__PURE__ */ e("svg", { style: { position: "absolute", width: 0, height: 0 }, "aria-hidden": !0, children: /* @__PURE__ */ e("clipPath", { id: "animal-modal-clip", clipPathUnits: "objectBoundingBox", children: /* @__PURE__ */ e("path", { d: "M0.501,0.005 L0.501,0.005 L0.523,0.005 L0.549,0.006 C0.704,0.01,0.796,0.017,0.825,0.027 L0.827,0.028 C0.872,0.045,0.939,0.044,0.978,0.17 C1,0.254,1,0.365,0.99,0.505 L0.988,0.513 C0.979,0.558,0.971,0.598,0.965,0.633 C0.956,0.689,0.979,0.77,0.964,0.865 C0.953,0.928,0.921,0.966,0.869,0.979 C0.821,0.986,0.773,0.992,0.726,0.995 L0.712,0.996 L0.694,0.997 C0.648,1,0.586,1,0.507,1 L0.501,1 L0.464,1 C0.385,1,0.325,0.998,0.283,0.995 C0.234,0.992,0.184,0.987,0.133,0.979 C0.081,0.966,0.05,0.928,0.039,0.865 C0.023,0.77,0.047,0.689,0.037,0.633 C0.031,0.595,0.023,0.552,0.013,0.505 C-0.006,0.365,-0.002,0.254,0.024,0.17 C0.064,0.045,0.13,0.045,0.174,0.028 L0.175,0.028 C0.204,0.017,0.303,0.009,0.474,0.005 L0.501,0.005" }) }) }), Ka = ({
  open: a,
  title: i,
  width: t = 520,
  maskClosable: o = !0,
  footer: c,
  onClose: n,
  onOk: l,
  children: s,
  className: r,
  typeSpeed: f = 80,
  typewriter: y = !0
}) => {
  const [k, u] = L(0);
  S(() => {
    a && u((C) => C + 1);
  }, [a]), S(() => {
    if (!a) return;
    const C = (P) => {
      P.key === "Escape" && n?.();
    };
    return document.addEventListener("keydown", C), () => document.removeEventListener("keydown", C);
  }, [a, n]), S(() => (a && (document.body.style.overflow = "hidden"), () => {
    document.body.style.overflow = "";
  }), [a]);
  const p = R(() => {
    o && n?.();
  }, [o, n]), m = R((C) => {
    C.stopPropagation();
  }, []);
  if (!a) return null;
  const v = /* @__PURE__ */ b(X, { children: [
    /* @__PURE__ */ e(q, { type: "primary", onClick: n, children: "取消" }),
    /* @__PURE__ */ e(q, { type: "primary", onClick: l, children: "确定" })
  ] }), d = /* @__PURE__ */ e(z, { children: /* @__PURE__ */ e("div", { className: T.mask, onClick: p, children: /* @__PURE__ */ b(
    "div",
    {
      className: [T.modal, r].filter(Boolean).join(" "),
      style: { width: t },
      onClick: m,
      role: "dialog",
      "aria-modal": "true",
      children: [
        /* @__PURE__ */ e(Ga, {}),
        /* @__PURE__ */ b("div", { className: T.modalClipped, children: [
          i && /* @__PURE__ */ e("div", { className: T.header, children: i && /* @__PURE__ */ e("div", { className: T.title, children: i }) }),
          /* @__PURE__ */ e("div", { className: T.body, children: y ? /* @__PURE__ */ e(Q, { speed: f, trigger: k, children: s }) : s }),
          c !== null && /* @__PURE__ */ e("div", { className: T.footer, children: c === void 0 ? v : c })
        ] })
      ]
    }
  ) }) });
  return na(d, document.body);
};
Ka.displayName = "Modal";
const Ua = "animal-card-DJ515", Za = "animal-card-title-PL-WB", Xa = "animal-card-dashed-fqnm1", Ja = "animal-card-app-pink-3Eo8d", Qa = "animal-card-purple-Gq7sR", _a = "animal-card-app-blue-GHH1a", an = "animal-card-app-yellow-DWB-L", nn = "animal-card-app-orange-V5yD7", en = "animal-card-app-teal-R5kFG", tn = "animal-card-app-green-xlNUs", on = "animal-card-app-red-DzE0l", cn = "animal-card-lime-green-KiJbe", ln = "animal-card-yellow-green-4c-lF", sn = "animal-card-brown-hLCrx", rn = "animal-card-warm-peach-pink-RlSVo", H = {
  card: Ua,
  "card-title": "animal-card-title-PL-WB",
  cardTitle: Za,
  "card-dashed": "animal-card-dashed-fqnm1",
  cardDashed: Xa,
  "card-app-pink": "animal-card-app-pink-3Eo8d",
  cardAppPink: Ja,
  "card-purple": "animal-card-purple-Gq7sR",
  cardPurple: Qa,
  "card-app-blue": "animal-card-app-blue-GHH1a",
  cardAppBlue: _a,
  "card-app-yellow": "animal-card-app-yellow-DWB-L",
  cardAppYellow: an,
  "card-app-orange": "animal-card-app-orange-V5yD7",
  cardAppOrange: nn,
  "card-app-teal": "animal-card-app-teal-R5kFG",
  cardAppTeal: en,
  "card-app-green": "animal-card-app-green-xlNUs",
  cardAppGreen: tn,
  "card-app-red": "animal-card-app-red-DzE0l",
  cardAppRed: on,
  "card-lime-green": "animal-card-lime-green-KiJbe",
  cardLimeGreen: cn,
  "card-yellow-green": "animal-card-yellow-green-4c-lF",
  cardYellowGreen: ln,
  "card-brown": "animal-card-brown-hLCrx",
  cardBrown: sn,
  "card-warm-peach-pink": "animal-card-warm-peach-pink-RlSVo",
  cardWarmPeachPink: rn
}, dn = ({
  type: a = "default",
  color: i = "default",
  children: t,
  className: o,
  style: c,
  ...n
}) => {
  const l = [
    H.card,
    a === "title" && H["card-title"],
    a === "dashed" && H["card-dashed"],
    i !== "default" && H[`card-${i}`],
    o
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ e(
    "div",
    {
      className: l,
      style: c,
      ...n,
      children: t
    }
  );
};
dn.displayName = "Card";
const mn = "animal-footer-5DMOR", pn = "animal-tree-KZ8hX", G = {
  footer: mn,
  tree: pn
}, hn = ({
  type: a = "sea",
  className: i,
  style: t
}) => {
  const o = [G.footer, G[a], i].filter(Boolean).join(" ");
  return /* @__PURE__ */ e("div", { className: o, style: t });
};
hn.displayName = "Footer";
const un = "animal-faqCard-5jlOE", bn = "animal-disabled-9DAjm", fn = "animal-questionHeader-iqP87", gn = "animal-questionIcon-xRkdp", wn = "animal-expanded-a8Eze", yn = "animal-questionText-jVpUv", Cn = "animal-leafDecoration-RoK4H", kn = "animal-answerWrapper-tzFtx", vn = "animal-answerContent-ppsk-", I = {
  faqCard: un,
  disabled: bn,
  questionHeader: fn,
  questionIcon: gn,
  expanded: wn,
  questionText: yn,
  leafDecoration: Cn,
  answerWrapper: kn,
  answerContent: vn
}, Nn = ({
  question: a,
  answer: i,
  defaultExpanded: t = !1,
  disabled: o = !1,
  className: c,
  style: n
}) => {
  const [l, s] = L(t), r = () => {
    o || s(!l);
  }, f = [
    I.faqCard,
    l && I.expanded,
    o && I.disabled,
    c
  ].filter(Boolean).join(" ");
  return /* @__PURE__ */ b("div", { className: f, style: n, children: [
    /* @__PURE__ */ b(
      "button",
      {
        className: I.questionHeader,
        onClick: r,
        disabled: o,
        "aria-expanded": l,
        children: [
          /* @__PURE__ */ e("span", { className: I.questionIcon, children: l ? "−" : "+" }),
          /* @__PURE__ */ e("span", { className: I.questionText, children: a }),
          /* @__PURE__ */ e("span", { className: I.leafDecoration, children: /* @__PURE__ */ e("svg", { viewBox: "0 0 24 24", width: "20", height: "20", children: /* @__PURE__ */ e(
            "path",
            {
              fill: "currentColor",
              d: "M17,8C8,10 5.9,16.17 3.82,21.34L5.71,22L6.66,19.7C7.14,19.87 7.64,20 8,20C19,20 22,3 22,3C21,5 14,5.25 9,6.25C4,7.25 2,11.5 2,13.5C2,15.5 3.75,17.25 3.75,17.25C7,8 17,8 17,8Z"
            }
          ) }) })
        ]
      }
    ),
    /* @__PURE__ */ e("div", { className: I.answerWrapper, children: /* @__PURE__ */ e("div", { className: I.answerContent, children: i }) })
  ] });
};
Nn.displayName = "Collapse";
const Ln = "animal-acDatetime-hVKh7", Dn = "animal-acDate-rhO3k", xn = "animal-acWeekday-bxDHR", In = "animal-acMonthday-1jUmX", Sn = "animal-acTime-S-twb", $n = "animal-acColon-g4vuJ", E = {
  acDatetime: Ln,
  acDate: Dn,
  acWeekday: xn,
  acMonthday: In,
  acTime: Sn,
  acColon: $n
}, Mn = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], Bn = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"], An = ({ className: a }) => {
  const [i, t] = L(/* @__PURE__ */ new Date());
  return S(() => {
    const o = setInterval(() => t(/* @__PURE__ */ new Date()), 1e3);
    return () => clearInterval(o);
  }, []), /* @__PURE__ */ e(z, { children: /* @__PURE__ */ b("div", { className: `${E.acDatetime} ${a || ""}`, children: [
    /* @__PURE__ */ b("div", { className: E.acDate, children: [
      /* @__PURE__ */ e("span", { className: E.acWeekday, children: Mn[i.getDay()] }),
      /* @__PURE__ */ b("span", { className: E.acMonthday, children: [
        Bn[i.getMonth()],
        " ",
        i.getDate()
      ] })
    ] }),
    /* @__PURE__ */ b("div", { className: E.acTime, children: [
      i.getHours().toString().padStart(2, "0"),
      /* @__PURE__ */ e("span", { className: E.acColon, children: ":" }),
      i.getMinutes().toString().padStart(2, "0")
    ] })
  ] }) });
};
An.displayName = "Time";
const Tn = "animal-phoneContainer-49yKw", En = "animal-phone-bXgSa", Rn = "animal-statusBar-Vb96q", Pn = "animal-time-cokU-", Hn = "animal-signalIcons-TuL3l", Wn = "animal-screenContent-oyJF-", On = "animal-homeScreen-no2yS", jn = "animal-grasswave-l6cyA", Fn = "animal-dateDisplay-cnvu8", qn = "animal-dateDisplayHeader-32i0J", Yn = "animal-blink--R0wf", Vn = "animal-dayText-7NEMb", zn = "animal-appsGrid-opTs8", Gn = "animal-appItem-1NmDt", Kn = "animal-appIcon-tTPIg", Un = "animal-iconBounce-n4Boc", Zn = "animal-appIconOffset-b5SZG", Xn = "animal-iconBounceOffset-XfsXI", Jn = "animal-appItemOffset-FEStg", Qn = "animal-iconCamera-c-grH", _n = "animal-iconMiles-L9Cow", ae = "animal-iconCritterpedia-okU95", ne = "animal-iconDiy-gZgxn", ee = "animal-iconDesign-YvE9z", te = "animal-iconMap-xmfHe", ie = "animal-iconVariant-k-kkB", oe = "animal-iconHelicopter-HV-TH", ce = "animal-iconChat-TVZXY", le = "animal-iconWifi-RoLJH", se = "animal-iconLocation-YfhoK", re = "animal-iconPage-wMmBC", de = "animal-badge-drGmk", me = "animal-pageIndicator-mQ-Pg", w = {
  phoneContainer: Tn,
  phone: En,
  statusBar: Rn,
  time: Pn,
  signalIcons: Hn,
  screenContent: Wn,
  homeScreen: On,
  grasswave: jn,
  dateDisplay: Fn,
  dateDisplayHeader: qn,
  blink: Yn,
  dayText: Vn,
  appsGrid: zn,
  appItem: Gn,
  appIcon: Kn,
  iconBounce: Un,
  appIconOffset: Zn,
  iconBounceOffset: Xn,
  appItemOffset: Jn,
  iconCamera: Qn,
  iconMiles: _n,
  iconCritterpedia: ae,
  iconDiy: ne,
  iconDesign: ee,
  iconMap: te,
  iconVariant: ie,
  iconHelicopter: oe,
  iconChat: ce,
  iconWifi: le,
  iconLocation: se,
  iconPage: re,
  badge: de,
  pageIndicator: me
}, pe = [
  { id: "camera", iconClass: "iconCamera", color: "#B77DEE", hasNewMessage: !0 },
  { id: "app", iconClass: "iconMiles", color: "#889DF0", offset: !0 },
  { id: "critterpedia", iconClass: "iconCritterpedia", color: "#F7CD67" },
  { id: "diy", iconClass: "iconDiy", color: "#E59266" },
  { id: "shopping", iconClass: "iconDesign", color: "#F8A6B2" },
  { id: "variant", iconClass: "iconMap", color: "#82D5BB", hasNewMessage: !0 },
  { id: "design", iconClass: "iconVariant", color: "#8AC68A" },
  { id: "map", iconClass: "iconHelicopter", color: "#FC736D" },
  { id: "chat", iconClass: "iconChat", color: "#D1DA49" }
], $t = ({ className: a }) => {
  const [i, t] = L(/* @__PURE__ */ new Date());
  S(() => {
    const r = setInterval(() => {
      t(/* @__PURE__ */ new Date());
    }, 1e3);
    return () => clearInterval(r);
  }, []);
  const o = i.getHours(), c = i.getMinutes(), n = o >= 12 ? "PM" : "AM", l = o % 12 || 12, s = c.toString().padStart(2, "0");
  return /* @__PURE__ */ e("div", { className: `${w.phoneContainer} ${a || ""}`, children: /* @__PURE__ */ e("div", { className: w.phone, children: /* @__PURE__ */ e("div", { className: w.screenContent, children: /* @__PURE__ */ b("div", { className: w.homeScreen, children: [
    /* @__PURE__ */ b("div", { className: w.dateDisplay, children: [
      /* @__PURE__ */ b("div", { className: w.dateDisplayHeader, children: [
        /* @__PURE__ */ e("span", { className: w.iconWifi }),
        /* @__PURE__ */ b("div", { children: [
          l,
          /* @__PURE__ */ e("span", { className: w.blink, children: ":" }),
          s,
          n
        ] }),
        /* @__PURE__ */ e("span", { className: w.iconLocation })
      ] }),
      /* @__PURE__ */ e("div", { className: w.dayText, children: "Welcome!" })
    ] }),
    /* @__PURE__ */ e("div", { className: w.appsGrid, children: pe.map((r) => /* @__PURE__ */ b(
      "div",
      {
        className: `${w.appItem} ${r.offset ? w.appItemOffset : ""}`,
        style: { backgroundColor: r.color },
        children: [
          r.hasNewMessage && /* @__PURE__ */ e("span", { className: w.badge }),
          /* @__PURE__ */ e(
            "span",
            {
              className: `${w.appIcon} ${w[r.iconClass]} ${r.offset ? w.appIconOffset : ""}`
            }
          )
        ]
      },
      r.id
    )) }),
    /* @__PURE__ */ e("div", { className: w.pageIndicator, children: /* @__PURE__ */ e("span", { className: w.iconPage }) })
  ] }) }) }) });
}, he = "animal-divider-ZBhpE", ue = "animal-line-teal-Z80hb", be = "animal-line-white-VDY1E", fe = "animal-line-yellow-ifvJz", ge = "animal-wave-yellow-AO-io", K = {
  divider: he,
  "line-teal": "animal-line-teal-Z80hb",
  lineTeal: ue,
  "line-white": "animal-line-white-VDY1E",
  lineWhite: be,
  "line-yellow": "animal-line-yellow-ifvJz",
  lineYellow: fe,
  "wave-yellow": "animal-wave-yellow-AO-io",
  waveYellow: ge
}, we = ({ type: a = "line-brown", className: i, style: t }) => {
  const o = [K.divider, K[a], i].filter(Boolean).join(" ");
  return /* @__PURE__ */ e("div", { className: o, style: t });
};
we.displayName = "Divider";
const ye = "animal-icon-wUFR-", Ce = "animal-iconBounce-Q1e3X", ke = "animal-icon-miles-oVFnM", ve = "animal-icon-camera-W9x4U", Ne = "animal-icon-chat-dfYOk", Le = "animal-icon-critterpedia-vDPSY", De = "animal-icon-design-8gBms", xe = "animal-icon-diy-AwPHs", Ie = "animal-icon-helicopter-w6ZL6", Se = "animal-icon-map-ogQk4", $e = "animal-icon-shopping-TSh2U", Me = "animal-icon-variant-E6wqy", Be = "animal-iconList-MRyTC", Ae = "animal-iconListItem-nTF3l", Te = "animal-iconListLabel-t2M3-", j = {
  icon: ye,
  "icon-bounce": "animal-icon-bounce-1MvWH",
  iconBounce: Ce,
  "icon-miles": "animal-icon-miles-oVFnM",
  iconMiles: ke,
  "icon-camera": "animal-icon-camera-W9x4U",
  iconCamera: ve,
  "icon-chat": "animal-icon-chat-dfYOk",
  iconChat: Ne,
  "icon-critterpedia": "animal-icon-critterpedia-vDPSY",
  iconCritterpedia: Le,
  "icon-design": "animal-icon-design-8gBms",
  iconDesign: De,
  "icon-diy": "animal-icon-diy-AwPHs",
  iconDiy: xe,
  "icon-helicopter": "animal-icon-helicopter-w6ZL6",
  iconHelicopter: Ie,
  "icon-map": "animal-icon-map-ogQk4",
  iconMap: Se,
  "icon-shopping": "animal-icon-shopping-TSh2U",
  iconShopping: $e,
  "icon-variant": "animal-icon-variant-E6wqy",
  iconVariant: Me,
  iconList: Be,
  iconListItem: Ae,
  iconListLabel: Te
}, Mt = ({
  name: a,
  size: i = 24,
  className: t,
  style: o,
  bounce: c = !1,
  ...n
}) => /* @__PURE__ */ e(
  "span",
  {
    className: `${j.icon} ${j[a]} ${c ? j["icon-bounce"] : ""} ${t || ""}`,
    style: {
      width: i,
      height: i,
      ...o
    },
    ...n
  }
), Bt = [
  { name: "icon-miles", label: "NookMiles" },
  { name: "icon-camera", label: "Camera" },
  { name: "icon-chat", label: "Chat" },
  { name: "icon-critterpedia", label: "Critterpedia" },
  { name: "icon-design", label: "Design" },
  { name: "icon-diy", label: "DIY" },
  { name: "icon-helicopter", label: "Helicopter" },
  { name: "icon-map", label: "Map" },
  { name: "icon-shopping", label: "Shopping" },
  { name: "icon-variant", label: "Variant" }
], Ee = "animal-wrapper-pbOTq", Re = "animal-trigger-2-Ahh", Pe = "animal-open-867xo", He = "animal-value-623RB", We = "animal-placeholder-vpC8E", Oe = "animal-arrow-xNbxA", je = "animal-dropdown-0l6jF", Fe = "animal-option-GEkLI", qe = "animal-active-jtSpt", Ye = "animal-hovered-N-cco", Ve = "animal-pillBar-O9cTb", ze = "animal-optionDot-FQtyw", Ge = "animal-disabled-Oy12C", N = {
  wrapper: Ee,
  trigger: Re,
  open: Pe,
  value: He,
  placeholder: We,
  arrow: Oe,
  dropdown: je,
  option: Fe,
  active: qe,
  hovered: Ye,
  pillBar: Ve,
  optionDot: ze,
  disabled: Ge
}, Ke = ({
  options: a,
  value: i,
  onChange: t,
  placeholder: o = "请选择",
  disabled: c = !1
}) => {
  const [n, l] = L(!1), [s, r] = L(null), [f, y] = L({}), [k, u] = L(!1), p = J(null), m = a.find((d) => d.key === i)?.label || o;
  S(() => {
    const d = (C) => {
      p.current && !p.current.contains(C.target) && (l(!1), u(!1));
    };
    return n && document.addEventListener("mousedown", d), () => document.removeEventListener("mousedown", d);
  }, [n]), S(() => {
    if (n && p.current) {
      const d = p.current.getBoundingClientRect(), C = window.innerWidth, P = window.innerHeight, $ = a.length * 44 + 24, h = {
        position: "absolute"
      };
      d.right + 200 > C ? (h.right = "100%", h.marginRight = "6px", h.left = "auto") : (h.left = "100%", h.marginLeft = "6px", h.right = "auto");
      const O = P - d.bottom, _ = d.top;
      O < $ && _ > O ? (h.top = "auto", h.bottom = "100%", h.marginBottom = "6px", delete h.transform) : O < $ || d.top < $ ? (h.top = "100%", h.marginTop = "6px", h.bottom = "auto", delete h.transform) : (h.top = "50%", h.transform = "translateY(-50%)", h.bottom = "auto"), y(h), requestAnimationFrame(() => {
        u(!0);
      });
    } else n || u(!1);
  }, [n, a.length]);
  const v = (d) => {
    t(d), l(!1), u(!1);
  };
  return /* @__PURE__ */ b(
    "div",
    {
      ref: p,
      className: `${N.wrapper} ${c ? N.disabled : ""}`,
      children: [
        /* @__PURE__ */ b(
          "div",
          {
            className: `${N.trigger} ${n ? N.open : ""}`,
            onClick: () => !c && l(!n),
            children: [
              /* @__PURE__ */ e("span", { className: i ? N.value : N.placeholder, children: m }),
              /* @__PURE__ */ e("span", { className: N.arrow, children: /* @__PURE__ */ e("svg", { width: "12", height: "12", viewBox: "0 0 12 12", fill: "none", children: /* @__PURE__ */ e("path", { d: "M3 4.5L6 7.5L9 4.5", stroke: "currentColor", strokeWidth: "1.5", strokeLinecap: "round", strokeLinejoin: "round" }) }) })
            ]
          }
        ),
        n && k && /* @__PURE__ */ e("div", { className: N.dropdown, style: f, children: a.map((d) => /* @__PURE__ */ b(
          "div",
          {
            className: `${N.option} ${i === d.key ? N.active : ""} ${s === d.key ? N.hovered : ""}`,
            onClick: () => v(d.key),
            onMouseEnter: () => r(d.key),
            onMouseLeave: () => r(null),
            children: [
              /* @__PURE__ */ e("span", { className: N.optionDot }),
              d.label,
              i === d.key && /* @__PURE__ */ e("div", { className: N.pillBar })
            ]
          },
          d.key
        )) })
      ]
    }
  );
};
Ke.displayName = "Select";
const Ue = "animal-tabs-I3QAo", Ze = "animal-tabList--fYUP", Xe = "animal-tabItem-Ehph4", Je = "animal-active-AoX4Y", Qe = "animal-tabIcon-Aiu-T", _e = "animal-tabLabel-bCauA", at = "animal-tabLeaf-1ud9k", nt = "animal-tabLeafStatic-52CX9", et = "animal-tabContent-zDlRq", tt = "animal-tabContentInner-Y5kRC", D = {
  tabs: Ue,
  tabList: Ze,
  tabItem: Xe,
  active: Je,
  tabIcon: Qe,
  tabLabel: _e,
  tabLeaf: at,
  tabLeafStatic: nt,
  tabContent: et,
  tabContentInner: tt
}, it = ({
  items: a,
  defaultActiveKey: i,
  activeKey: t,
  onChange: o,
  className: c,
  style: n,
  leafAnimation: l = !0
}) => {
  const [s, r] = L(
    i || a[0]?.key
  ), f = t !== void 0 ? t : s, y = (p) => {
    t === void 0 && r(p), o?.(p);
  }, k = a.find((p) => p.key === f), u = [D.tabs, c].filter(Boolean).join(" ");
  return /* @__PURE__ */ b("div", { className: u, style: n, children: [
    /* @__PURE__ */ e("div", { className: D.tabList, children: a.map((p) => {
      const m = p.key === f;
      return /* @__PURE__ */ b(
        "button",
        {
          className: `${D.tabItem} ${m ? D.active : ""}`,
          onClick: () => y(p.key),
          children: [
            /* @__PURE__ */ e("span", { className: D.tabIcon, children: m ? "●" : "○" }),
            /* @__PURE__ */ e("span", { className: D.tabLabel, children: p.label }),
            m && /* @__PURE__ */ e("img", { src: ea, alt: "", className: `${D.tabLeaf} ${l ? "" : D.tabLeafStatic}` })
          ]
        },
        p.key
      );
    }) }),
    /* @__PURE__ */ e("div", { className: D.tabContent, children: /* @__PURE__ */ e("div", { className: D.tabContentInner, children: k?.children }) })
  ] });
};
it.displayName = "Tabs";
const ot = "animal-checkboxGroup-q3P6l", ct = "animal-horizontal-fBBSw", lt = "animal-vertical-klL45", st = "animal-checkboxItem-DxKFM", rt = "animal-disabled-zDC-1", dt = "animal-box-bIDkg", mt = "animal-label-L0ila", pt = "animal-small-2wCHI", ht = "animal-checkmark-Ub21Y", ut = "animal-middle-VUB-E", bt = "animal-large-QQxPe", ft = "animal-checked-S7gPe", gt = "animal-animal-checkbox-pop-0cWeC", wt = "animal-groupDisabled-LvBFd", x = {
  checkboxGroup: ot,
  horizontal: ct,
  vertical: lt,
  checkboxItem: st,
  disabled: rt,
  box: dt,
  label: mt,
  small: pt,
  checkmark: ht,
  middle: ut,
  large: bt,
  checked: ft,
  "animal-checkbox-pop": "animal-animal-checkbox-pop-0cWeC",
  animalCheckboxPop: gt,
  groupDisabled: wt
};
function yt(a) {
  return a && a.__esModule && Object.prototype.hasOwnProperty.call(a, "default") ? a.default : a;
}
var F = { exports: {} };
var U;
function Ct() {
  return U || (U = 1, (function(a) {
    (function() {
      var i = {}.hasOwnProperty;
      function t() {
        for (var n = "", l = 0; l < arguments.length; l++) {
          var s = arguments[l];
          s && (n = c(n, o(s)));
        }
        return n;
      }
      function o(n) {
        if (typeof n == "string" || typeof n == "number")
          return n;
        if (typeof n != "object")
          return "";
        if (Array.isArray(n))
          return t.apply(null, n);
        if (n.toString !== Object.prototype.toString && !n.toString.toString().includes("[native code]"))
          return n.toString();
        var l = "";
        for (var s in n)
          i.call(n, s) && n[s] && (l = c(l, s));
        return l;
      }
      function c(n, l) {
        return l ? n ? n + " " + l : n + l : n;
      }
      a.exports ? (t.default = t, a.exports = t) : window.classNames = t;
    })();
  })(F)), F.exports;
}
var kt = Ct();
const Z = /* @__PURE__ */ yt(kt), vt = ({
  value: a,
  defaultValue: i = [],
  options: t,
  size: o = "middle",
  disabled: c = !1,
  direction: n = "horizontal",
  onChange: l,
  className: s,
  style: r
}) => {
  const [f, y] = L(i), k = a !== void 0, u = k ? a : f, p = R(
    (m, v) => {
      if (c || v) return;
      const d = u.includes(m) ? u.filter((C) => C !== m) : [...u, m];
      k || y(d), l?.(d);
    },
    [c, u, k, l]
  );
  return /* @__PURE__ */ e(
    "div",
    {
      className: Z(
        x.checkboxGroup,
        x[n],
        { [x.groupDisabled]: c },
        s
      ),
      style: r,
      children: t.map((m) => {
        const v = u.includes(m.value), d = c || m.disabled;
        return /* @__PURE__ */ b(
          "label",
          {
            className: Z(
              x.checkboxItem,
              x[o],
              {
                [x.checked]: v,
                [x.disabled]: d
              }
            ),
            onClick: () => p(m.value, m.disabled),
            children: [
              /* @__PURE__ */ e(
                "span",
                {
                  className: x.box,
                  role: "checkbox",
                  "aria-checked": v,
                  tabIndex: d ? -1 : 0,
                  onKeyDown: (C) => {
                    (C.key === " " || C.key === "Enter") && (C.preventDefault(), p(m.value, m.disabled));
                  },
                  children: v && /* @__PURE__ */ e("span", { className: x.checkmark, children: "✓" })
                }
              ),
              /* @__PURE__ */ e("span", { className: x.label, children: m.label })
            ]
          },
          String(m.value)
        );
      })
    }
  );
};
vt.displayName = "Checkbox";
const g = {
  comment: "#6b5e50",
  string: "#a8d4a0",
  keyword: "#d4a0e0",
  react: "#e06c75",
  component: "#80c0e0",
  func: "#61afef",
  prop: "#e8c87a",
  jsx: "#f0a870",
  operator: "#d4b896",
  number: "#a8d4a0",
  default: "#e8d5bc"
}, Nt = {
  padding: "20px 24px",
  background: "#2b2118",
  border: "1px solid #3d3028",
  borderRadius: 20,
  fontSize: 14,
  lineHeight: 1.7,
  fontFamily: "'SF Mono', 'Fira Code', 'Cascadia Code', Consolas, monospace",
  fontWeight: 600,
  color: "#e8d5bc",
  whiteSpace: "pre",
  overflow: "auto",
  tabSize: 4
}, Lt = (a) => {
  const i = [], t = (n, l) => {
    let s;
    const r = new RegExp(n.source, n.flags.includes("g") ? n.flags : n.flags + "g");
    for (; (s = r.exec(a)) !== null; )
      i.push({
        start: s.index,
        end: s.index + s[0].length,
        color: l
      });
  };
  t(/\/\*[\s\S]*?\*\//g, g.comment), t(/\/\/.*$/gm, g.comment), t(/`[^`]*`/g, g.string), t(/"[^"]*"/g, g.string), t(/'[^']*'/g, g.string), t(/<\/?[A-Z][\w.$]*/g, g.jsx), t(/<\/?[a-z][\w-]*/g, g.jsx), t(/\/?>/g, g.jsx), t(/\b(React|useState|useEffect|useCallback|useMemo|useRef|useContext|useReducer|useLayoutEffect|useImperativeHandle|useDebugValue|createContext|createElement|cloneElement|Fragment|Suspense|lazy|memo|forwardRef|useId|FC|ReactNode|ReactElement|CSSProperties)\b/g, g.react), t(/\b(true|false)\b/g, g.keyword), t(/\b(null|undefined|void|NaN|Infinity)\b/gi, g.keyword), t(/\b\d+\.?\d*\b/g, g.number), t(/\b(import|from|as|export|default|const|let|var|function|return|if|else|for|while|switch|case|break|continue|try|catch|throw|finally|new|typeof|instanceof|async|await|type|interface)\b/gi, g.keyword), t(/\b[A-Z][a-zA-Z0-9_$]*\b/g, g.component), t(/\b[a-z][a-zA-Z0-9_$]*\s*(?=\()/g, g.func), t(/\b[a-zA-Z_$][\w$]*\s*(?==)/g, g.prop), t(/>|===|!==|==|!=|<=|>=|&&|\|\||[+\-*/%=<>!&|^~?:]/g, g.operator), t(/[{}[\]();,]/g, g.operator), i.sort((n, l) => n.start - l.start);
  const o = [];
  let c = 0;
  for (const n of i)
    n.start < c || (n.start > c && o.push(/* @__PURE__ */ e("span", { style: { color: g.default }, children: a.slice(c, n.start) }, `t${c}`)), o.push(/* @__PURE__ */ e("span", { style: { color: n.color }, children: a.slice(n.start, n.end) }, `s${n.start}`)), c = n.end);
  return c < a.length && o.push(/* @__PURE__ */ e("span", { style: { color: g.default }, children: a.slice(c) }, `e${c}`)), o;
}, At = ({ code: a, style: i, className: t }) => /* @__PURE__ */ e("pre", { style: { ...Nt, ...i }, className: t, children: Lt(a) });
export {
  q as Button,
  dn as Card,
  vt as Checkbox,
  At as CodeBlock,
  Nn as Collapse,
  z as Cursor,
  we as Divider,
  hn as Footer,
  Bt as ICON_LIST,
  Mt as Icon,
  $a as Input,
  Ka as Modal,
  $t as Phone,
  Ke as Select,
  Wa as Switch,
  it as Tabs,
  An as Time,
  Q as Typewriter
};
